apicast/src/oauth/keycloak.lua

```diff
@@ -0,0 +1,243 @@
+local setmetatable = setmetatable
+local len = string.len
+local http_ng = require "resty.http_ng"
+local resty_url = require 'resty.url'
+local jwt = require 'resty.jwt'
+local cjson = require 'cjson'
+
+local _M = {
+  _VERSION = '0.1'
+}
+local mt = {
+  __index = _M
+}
+
+-- Required params for each grant type and response type.
+_M.params = {
+  grant_type = {
+    ['authorization_code'] = {'client_id','redirect_uri','code'},
+    ['password'] = {'client_id','client_secret','username','password'},
+    ['client_credentials'] = {'client_id','client_secret'}
+  },
+  response_type = {
+    ['code'] = {'client_id','redirect_uri'},
+    ['token'] = {'client_id','redirect_uri'}
+  }
+}
+
+function _M.init(endpoint)
+  _M.configured = endpoint
+
+  local config = { endpoint = endpoint}
+  if _M.configured then
+    _M.configuration = config
+  end
+end
+
+local function get_public_key(http_client, endpoint)
+  if not http_client then
+    return nil, 'not initialized'
+  end
+
+  local res = http_client.get(endpoint)
+  local key
+  if res.status == 200 then
+    local json = cjson.decode(res.body)
+    key = json.public_key
+  end
+
+  if not key then
+    return nil, 'missing key'
+  end
+
+  local formatted_key = "-----BEGIN PUBLIC KEY-----\n"
+  local key_len = len(key)
+  for i=1,key_len,64 do
+    formatted_key = formatted_key..string.sub(key, i, i+63).."\n"
```

Comment:
Where is that documented BTW ?

Because it uses AES from https://github.com/openresty/lua-resty-string and it does not mention anything. 

Could that be what OpenSSL requires?